### DISCUSSION TITLE
Compatibility with Alexa

### DISCUSSION THREAD

**Discussion by fritut08** [krahabb/meross_lan#223] [Q&A]
Hey, I have not yet installed this integration but am excited to try it out. I have one question regarding the compatibility with Amazon Alexa, though. If I use this integration will I still be able to switch the Meross plug on and off using my Echo Dot? Does anyone know whether the default Meross skill talks directly to the Meross plug through http or uses the Meross cloud?

**Comment by krahabb** [comment]
Hello @fritut08,
When you pair your devices with the Meross app, the Alexa support works by talking with the Meross cloud (I'm not 100% sure since I've never experienced it but this is how it works 99% of the times).
If you use meross_lan side by side with the Meross app (i.e. you leave your devices paired with the app but you also add them into HA/meross_lan when they're discovered or by manually adding them by IP address) every feature of the official Meross cloud will still work since meross_lan is just 'side communicating' locally over HTTP directly to the devices and is not 'disrupting' the cloud functions.

This is the 'easiest', default, path. 

Keep in mind that, if you integrate Alexa with HA too, you can control any of your HA entities through the HA <-> Alexa integration (the same works for Google Assistant) even if they're not supported by their original manufacturers. This means you could have 2 different paths for using Alexa with your Meross devices.

**Reply by fritut08** [reply]
@krahabb Thanks for the reply! So, if I want my Alexa to control the Meross devices without them being connected to the Meross cloud, there will be no way around coupling Alexa with HA either using the HA cloud or my own setup (as described [here](https://www.home-assistant.io/integrations/alexa.smart_home/)).

I had noticed that controlling the Meross devices with the Meross app worked even if I denied them internet access through my router's settings. So, the app must fallback to local http requests if the devices are not connected to the cloud. Alexa was not able to reach the device but I had hoped there would be some way to make Alexa send http requests.

**Reply by krahabb** [reply]
> @krahabb Thanks for the reply! So, if I want my Alexa to control the Meross devices without them being connected to the Meross cloud, there will be no way around coupling Alexa with HA either using the HA cloud or my own setup (as described [here](https://www.home-assistant.io/integrations/alexa.smart_home/)).

Yes, Alexa works on the cloud since speak recognition is actually performed on dedicated servers (that's true for GoogleAssistant too even tho I've read Google is trying to port some recognition tasks locally on their Pixel devices for instance..but that's an edge feature I guess)
So, Alexa never talks to local devices in my opinion. 
Things could change with the incoming 'Matter' standard though. Maybe the Alexa speaker itself will be able to offload the voice recognition tasks to its own cloud and then complete the action (once recognized) locally by talking to supported devices through the Matter standard
